### PR TITLE
Fix gRPC context propagation

### PR DESCRIPTION
**Pull request by alesj**
This fixes #32046 issue.

### CODE REVIEW

**Review by cescoffier — APPROVED** [review]

**Comment by alesj** [comment]
@cescoffier wait with the merge ... let me test this on blocking methods as well ...

**Comment by alesj** [comment]
@cescoffier OK, added a blocking test, ad it worked ootb

File: extensions/grpc/runtime/src/main/java/io/quarkus/grpc/runtime/supports/blocking/BlockingServerInterceptor.java [hunk 1]
```diff
@@ -10,6 +10,9 @@
 import java.util.function.Consumer;
 import java.util.function.Function;
 
+import org.slf4j.Logger;
```
Comment (cescoffier) [inline extensions/grpc/runtime/src/main/java/io/quarkus/grpc/runtime/supports/blocking/BlockingServerInterceptor.java:13]: Should use jboss logging

**Review by cescoffier — CHANGES_REQUESTED** [review]
LGTM except the logger api

**Comment by cescoffier** [comment]
@alesj can you rebase, just to be sure?

**Comment by geoand** [comment]
Whats the status of this? If it's ready to be merged, please rebase once more and then add the `waiting-for-ci` label

**Comment by cescoffier** [comment]
No it's not. There is a race somewhere.

**Comment by geoand** [comment]
Understood, thanks

**Comment by cescoffier** [comment]
@alesj do you believe we can make the race more easily reproducible?

**Comment by alesj** [comment]
It’s been a while, but I remember I saw it quite often.
As all race conditions, it’s not easy to reproduce though. :-)

**Comment by cescoffier** [comment]
@alesj I just pushed a one-line-fix. When creating a new context, you need to attach it explicitly. Thanks to the duplicated context handling, it will attach it to the duplicated context of the call.

**Review by cescoffier — APPROVED** [review]

**Pull request merged**